### DISCUSSION TITLE
Add extra holiday days?

### DISCUSSION THREAD

**Discussion by JohnBull-1975** [vacanza/holidays#1204] [Ideas]
Hi,
Python-Holidays is great integrated into Home Assistant!
However, we have a few days which we always have as free from work apart from the public holidays for Sweden (SE)
Is it possible to modify and add custom days in some way?
For example we are always free on the Friday after Assension Day and we are always free between x-mas and New Years.
A more complicated one is that we are free on the day before Swedish National day June 6 if it's a Monday and the day after if that's a Friday

It would be great to be able to alter this and avoid lights going on to early in the mornings these days without manually inactivateing automation!

**Comment by arkid15r** [comment]
Hi @JohnBull-1975,

there is a way to add custom holidays for the HA workday sensor. It used to be a config based and was recently moved to integrations. You need just add those dates into Workday configuration manually and live happily those upcoming mornings: use "Add holidays dropdown" at Workday settings modal (see https://www.home-assistant.io/integrations/workday/ for more details).

In case that's something you've been already aware of I can only assume you're looking for a way to add those "semi-holidays" into the PH directly which is not supported at this moment. However, as we see growing demand on non-public holidays support, that may change in the future.

Let me know if you feel a part of your question left uncovered or you'd like some clarification.

Thank you for your question!

**Reply by JohnBull-1975** [reply]
Hi and thanks for replying
However, as far as I can see, it's only possible to add custom holidays according to hard coded dates, e.g. YYYY-MM-DD and that will solve the period between x-mas and new years. 

But it is obviously not possible at the moment to add for example "Assension Day + 1" or "Monday June 5" and "Friday June 7"
That would have made this function complete!

**Reply by arkid15r** [reply]
Right, now I see your concern. You're correct -- all the custom holidays need to be entered manually. There is no flexibility there.
Even if PH supported "extra" stuff (not only official/public holidays) it'd require a deeper integration w/ Workday sensor.

I'm sorry that PH + HA can't satisfy your use-case fully at the moment. However, you as a user have a good workaround and we as developers/maintainers have some food for thoughts based on the use-case you've described.

My understanding is that PH still has the [correct data](https://publicholidays.se/2023-dates/) for Sweden and no action is needed even though in fact people usually have more non-working days.

Thank you!

**Reply by JohnBull-1975** [reply]
Hi again!
Correct, everything is correct for Sweden!
It's just a flexible opportunity for custom holidays that would be great.
